README.md

```diff
@@ -18,6 +18,8 @@ Google Cloud configuration:
 
 Check that you have the `make` utility installed, and if not (which is unlikely), install it using your system package manager.
 
+Ensure that you have `java` version 8 installed on your system (also known as java 1.8).
```

Comment:
Oh maybe it doesn't need to just be version 8. I was having an issue running pytest with Java 20. @tskir and I talked about it and downgrading to version 8 solved my error (which was `Py4JJavaError(
py4j.protocol.Py4JJavaError: An error occurred while calling None.org.apache.spark.api.java.JavaSparkContext.`).

The PySpark documentation says 'PySpark requires Java version 1.8. 0 or the above version' so not sure why I had an issue with version 20. Maybe we can reject this change in the README for now? I can look more into if there is a specific requirement or if I just had something else java-related going wrong.

Reply:
I think what's happening here is that while PySpark declares it needs version 8 (and it indeed works on 8), it actually unofficially supports some higher versions as well. (Anecdotally, I encountered some other examples of software that require JRE 8 on paper but actually run fine on 10-11.) But evidently changes in version 20 are too large and it can't support that anymore.

I would definitely include some change to the README mentioning Java, because when someone has an incorrect version, it doesn't just say "Wrong Java version", it generates a _really_ long and cryptic traceback which is quite hard to make sense of.

I would revise the change to say something like: "PySpark requires Java version 8, a. k. a. 1.8, to run propertly. It may also work on higher versions up to at least 11, but will not work on high versions like Java 20. If you are encountering problems with initialising a Spark session, make sure you have the correct version of Java, ideally Java 8."

Reply:
Actually, on re-reading PySpark requirements, it doesn't look like it's version locked. So it probably will work on most versions, and the issue with 20 is that it's _too_ new (released in March 23), and Spark didn't have time to adapt. So then the revised revised statement would read:

> PySpark requires Java version 8 (a.k.a. 1.8) or above to work. However, it may not work with the most recent Java versions, like Java 20 at the time of writing this section (May 2023). If you are encountering problems with initialising a Spark session, try using Java 11 or Java 8.